omnisend/includes/SDK/V1/class-event.php

```diff
@@ -0,0 +1,145 @@
+<?php
+/**
+ * Omnisend Client
+ *
+ * @package OmnisendClient
+ */
+
+namespace Omnisend\SDK\V1;
+
+use Omnisend\SDK\V1\Contact;
+use WP_Error;
+
+defined( 'ABSPATH' ) || die( 'no direct access' );
+
+/**
+ * Omnisend Event class. It's should be used with Omnisend Client.
+ *
+ */
+class Event {
+
+
+	private $contact          = null;
+	private $event_name       = null;
+	private $event_time       = null;
+	private $origin           = null;
+	private array $properties = array();
+
+	/**
+	 * Validate event properties.
+	 *
+	 * TODO kas yra reuquired? ka reikia vailiduoti?
+	 * It ensures that all required properties are set
+	 *
+	 * @return WP_Error
+	 */
+	public function validate(): WP_Error {
+
+		$error = new WP_Error();
+		if ( $contact instanceof Contact ) {
+			$error->merge_from( $contact->validate() );
+		}
+
+		return $error;
+	}
+
+	/**
+	 * Sets event name.
+	 *
+	 * @param $event_name
+	 *
+	 * @return void
+	 */
+	public function set_event_name( $event_name ): void {
+		$this->event_name = $event_name;
+	}
+
+	/**
+	 * Sets event time.
+	 *
+	 * @param $event_time
+	 *
+	 * @return void
+	 */
+	public function set_event_time( $event_time ): void {
+		$this->event_time = $event_time;
```

Comment:
Would be good to validate. If you pass incorrect format api request will fail. Please add validation or remove this property at all.

Reply:
fixed,please check

Reply:
Value could be not only string, but number, bool, event array.